packages/next/src/server/config-shared.ts

```diff
@@ -740,8 +740,8 @@ export interface NextConfig extends Record<string, any> {
   rewrites?: () => Promise<
     | Rewrite[]
     | {
-        beforeFiles: Rewrite[]
-        afterFiles: Rewrite[]
+        beforeFiles?: Rewrite[]
+        afterFiles?: Rewrite[]
         fallback: Rewrite[]
```

Comment:
```suggestion
        fallback?: Rewrite[]
```
All of them should be optional